### PR TITLE
Improved Markdown formatting of Breaking changes documents

### PR DESCRIPTION
**Pull request by svick**
Not sure if release/dev16.4-preview1 is the right target branch for a PR that doesn't change any code, but targeting that branch just in case anyway.

### CODE REVIEW

**Comment by JoeRobich** [comment]
Sorry, I should have retargeted these PRs before removing that branch. Repushed branch and will reopen and retarget.

File: docs/compilers/CSharp/Compiler Breaking Changes - VS2015.md [hunk 1]
```diff
@@ -146,49 +146,52 @@ This page tracks known cases where the Roslyn compiler intentionally deviates fr
...
-        {
+    {
         throw null;
-        }
+    }
 }
 class Program
 {
     static void Main(string[] args)
         {
```
Comment (sharwell) [inline docs/compilers/CSharp/Compiler Breaking Changes - VS2015.md]: 💡 These braces could be outdented one level

File: docs/compilers/CSharp/Compiler Breaking Changes - VS2015.md [hunk 2]
```diff
@@ -146,49 +146,52 @@ This page tracks known cases where the Roslyn compiler intentionally deviates fr
...
+
 ### C# spec deviation: New Int32() treated as constant 0
 
-```
+```c#
             // DELIBERATE SPEC VIOLATION:
             // The spec does not allow "new int()" to be treated as a constant 
             // The native compiler treats "new int()" (and so on) as a "zero" constant expression, 
             // despite the fact that the specification does not include object creation expressions on the
             // list of legal constant expressions.
```
Comment (sharwell) [inline docs/compilers/CSharp/Compiler Breaking Changes - VS2015.md]: 💡 This entire block could be outdented all the way

File: docs/compilers/CSharp/Compiler Breaking Changes - VS2015.md [hunk 3]
```diff
@@ -284,43 +291,49 @@ class Program
...
+
 ### C# spec deviation: Static class as an interface method's parameter (C#)
 
 Dev10 allows a static class to be used as the type of an interface method parameter and an interface method return type.  This is disallowed by the language spec (10.1.1.3.1) but permitted by the Dev10 compiler.
 
 We follow suit in Roslyn so as not to break compatibility with (probably useless) existing programs.
+
 ### C# spec deviation: null??null and other null expressions
 
+```c#
```
Comment (sharwell) [inline docs/compilers/CSharp/Compiler Breaking Changes - VS2015.md]: For consistency with the rest of the document

```suggestion
``` cs
```

**Comment by svick** [comment]
@sharwell I have fixed the cases that you mentioned, and also few others in the same document.

**Pull request merged**